swan-lake/development-tutorials/additional-tools/scan-tool.md

```diff
@@ -85,6 +85,98 @@ To ignore a specific set of rules during the analysis, use the following command
\ No newline at end of file
+SonarQube is a popular open-source platform for continuous inspection of code quality. It provides static code analysis, code coverage, and other features to help developers maintain clean, maintainable codebases. The Ballerina scan tool can be integrated with SonarQube to publish static code analysis reports, enabling seamless integration into your CI/CD pipelines.
+
+This guide walks you through the process of configuring SonarQube and publishing Ballerina static code analysis reports.
+
+### Prerequisites
+- SonarQube 9.9 LTA Community Edition installed.
+- SonarScanner CLI 4.8.0 or later installed, and added to your system `PATH`.
+- [SonarQube Ballerina plugin](https://github.com/ballerina-platform/sonar-ballerina/packages/2580146), and [SonarQube platform plugin](https://github.com/ballerina-platform/sonar-ballerina/packages/2580149) downloaded.
+
+### Configure the SonarQube server
+
+1. Install [Java 17](https://adoptium.net/en-gb/temurin/releases/?version=17) in your machine.
+   - If Java 17 is not the default Java installation, override it:
+
+      **For Unix/macOS:**
+      ```bash
+      export SONAR_JAVA_PATH="path/to/java17_home/bin/java"
+      ```
+
+      **For Windows:**
+      ```bash
+      setx SONAR_JAVA_PATH="path\to\java17_home\bin\java"
+      ```
+
+2. Setup SonarQube 9.9 LTA
+   - Download SonarQube 9.9 LTA from [here](https://www.sonarsource.com/products/sonarqube/downloads/historical-downloads/).
+   - Extract the downloaded zip file.
+
+3. Add the [SonarQube Ballerina plugin](https://github.com/ballerina-platform/sonar-ballerina/packages/2580146).
+   - Download the latest Ballerina SonarQube plugin JAR.
+   - Place the JAR file into the `extensions/plugins/` directory of your SonarQube installation.
+
+4. Navigate to the appropriate `bin/<OS>/` directory and run:
+   ```bash
+   $ ./sonar.sh start
+   ```
+
+   You can access the SonarQube dashboard at http://localhost:9000 once the server is up.
+
+5. Create a new project in SonarQube.
+   - Log in to the SonarQube dashboard.
+   - Click on `Create Project`.
+   - Follow the prompts to set up your project.
+
+6. Install and Configure SonarScanner CLI.
+   - Download SonarScanner CLI from [here](https://docs.sonarsource.com/sonarqube-server/9.9/analyzing-source-code/scanners/sonarscanner/).
+   - Add it to your system `PATH`.
+   - Ensure `sonar.host.url` is set correctly (either via a properties file or CLI parameter).
+
+### Configure the Ballerina project
+
+1. Download the [SonarQube platform plugin](https://github.com/ballerina-platform/sonar-ballerina/packages/2580149).
+
+2. Create a `sonar-project.properties` file at the root of your Ballerina project with the following content:
+   ```properties
```

Comment:
`properties` is not supported by shiki (https://shiki.style/languages), which is the syntax highlighter used in b.io.  Check whether we can use an alternative language code.